cleanlab/classification.py

```diff
@@ -782,123 +790,160 @@ def find_label_issues(
         self.ps = value_counts(labels) / float(len(labels))
 
         self.clf_kwargs = clf_kwargs
-        self._process_label_issues_kwargs(self.find_label_issues_kwargs)
-        # self._process_label_issues_kwargs might set self.confident_joint. If so, we should use it.
-        if self.confident_joint is not None:
-            self.py, noise_matrix, inv_noise_matrix = estimate_latent(
-                confident_joint=self.confident_joint,
-                labels=labels,
-            )
-
-        # If needed, compute noise rates (probability of class-conditional mislabeling).
-        if noise_matrix is not None:
-            self.noise_matrix = noise_matrix
-            if inverse_noise_matrix is None:
+        if self.low_memory:
+            # If needed, compute P(label=k|x), denoted pred_probs (the predicted probabilities)
+            if pred_probs is None:
                 if self.verbose:
-                    print("Computing label noise estimates from provided noise matrix ...")
-                self.py, self.inverse_noise_matrix = compute_py_inv_noise_matrix(
-                    ps=self.ps,
-                    noise_matrix=self.noise_matrix,
+                    print(
+                        "Computing out of sample predicted probabilities via "
+                        f"{self.cv_n_folds}-fold cross validation. May take a while ..."
+                    )
+
+                pred_probs = estimate_cv_predicted_probabilities(
+                    X=X,
+                    labels=labels,
+                    clf=self.clf,
+                    cv_n_folds=self.cv_n_folds,
+                    seed=self.seed,
+                    clf_kwargs=self.clf_kwargs,
+                    validation_func=validation_func,
                 )
-        if inverse_noise_matrix is not None:
-            self.inverse_noise_matrix = inverse_noise_matrix
-            if noise_matrix is None:
-                if self.verbose:
-                    print("Computing label noise estimates from provided inverse noise matrix ...")
-                self.noise_matrix = compute_noise_matrix_from_inverse(
-                    ps=self.ps,
-                    inverse_noise_matrix=self.inverse_noise_matrix,
+
+            if self.verbose:
+                print("Using predicted probabilities to identify label issues ...")
+
+            arg_values = {
+                "thresholds": thresholds,
+                "noise_matrix": noise_matrix,
+                "inverst_noise_matrix": inverse_noise_matrix,
+                "find_label_issues_kwargs": self.find_label_issues_kwargs,
+            }
+            for arg_name, arg_val in arg_values.items():
+                if arg_val:
+                    warnings.warn(f"`{arg_name}` is not used when `low_memory=True`.")
+            label_issues_indices = find_label_issues_batched(labels, pred_probs)
```

Comment:
Can add `labels = labels_to_array(labels)` above this line

Reply:
Sorry, I am a little confused. 
*  In the [first comment](https://github.com/cleanlab/cleanlab/pull/791#discussion_r1286601954), you mention that we want to preserve the nested array format in labels. Doesn't this mean `labels = labels_to_array(labels)` cannot be used anytime before calling `compute_confident_joint`, because it doesn't preserve the nested structure?  If so, how else could I resolve this issue?
* Should `labels = labels_to_array(labels)` be added only for `find_label_issues_batched`? The unit test would still fail when `low_memory` is not True.

Reply:
Apologies if my comment was unclear, I took a closer look at the source code and here are some notes:

- The unittest you added actually caught a bug that was present even before the `low_memory` stuff was added - thanks for the great catch! 
  I was initially confused because I thought the `low_memory` addition introduced the bug hence I proposed the solution above.

- When I mentioned this above -
  > We should not do this as we intentionally want to preserve the nested array format in labels for some of the following function calls (eg. `get_confident_thresholds`).

  I mainly meant that we should not alter than function in `count.py` as it is used by many other functions in the library. In general, we should always be cautious about changing the data formats in these commonly used functions. 
  The biggest reason behind not adding this line of code in `count.py` is because that function also handles labels for multi-label classification which are formatted in nested list format. However, `CleanLearning.find_label_issues` currently only supports multi-class classification hence we do have more liberty to reformat these labels. Because we don't have the multi-label case to worry about, we can probably now move this `labels = labels_to_array(labels)` line before the if-else statements (perhaps as high as L770) to ensure it gets called in both cases, and also before the function calls that expect labels to be formatted properly.

Reply:
Apologies for the confusion regarding the unit-test! I should have been clearer about it.
Thank you for the detailed explanation regarding `labels = labels_to_array(labels)`, I will make the change in `CleanLearning.find_label_issues`.